include/sdlmain.h

```diff
@@ -89,6 +90,25 @@ enum PRIORITY_LEVELS {
 	PRIORITY_LEVEL_HIGHEST
 };
 
+struct ViewportSettings {
+	ViewportMode mode = {};
+
+	// Either parameter can be set in Fit mode (but not both at the
+	// same time), or none
+	struct {
+		std::optional<DosBox::Rect> limit_rectangle = {};
+		std::optional<float> desktop_percent        = {};
+	} fit = {};
```

Comment:
Is the `limit_rectangle` just the relative dimensions (`width` and `height` only) or do its `x` and `y` values also position the viewport on the display, in an absolute sense?

The ``limit_rectangle` name implies the latter; just wanted to double check!

Reply:
Just width & height. I just don't wanna go overboard with having a different `Size` type in addition to `Rect`, I think it's overkill. From code it becomes obvious what we're doing, and there's a limit how much info you can encode into names & types (unless you're happy to introduce tons of types 😅).

Reply:
yup - was only asking just for the name of the variable (`limit_rectangle`) -- if it's just dimensions, maybe it could be named `limit_dims` or `limit_dimensions`)?  

Rectange at least to me says that position is probably involved too :-)

Reply:
Yeah fair point, renaming.